test/test_linalg.py

```diff
@@ -825,23 +825,61 @@ def test_kron_errors_and_warnings(self, device, dtype):
     # as expected, according to the function's documentation
     @skipCUDAIfNoMagma
     def test_norm_dtype(self, device):
-        def run_test_case(input_size, ord, keepdim, from_dtype, to_dtype, compare_dtype):
+        def run_test_case(input_size, ord, keepdim, from_dtype, to_dtype):
+            # Determine the best dtype to use for comparisons between tensors
+            # of two different types
+            def get_compare_dtype(type0, type1):
+                types_32bit_based = [torch.float, torch.cfloat]
```

Comment:
can get_compare_dtype just be torch.promote_types()? 

https://pytorch.org/docs/master/generated/torch.promote_types.html?highlight=promote#torch.promote_types

Reply:
`torch.promote_types` doesn't seem to do what I want. It chooses the higher precision type and prefers complex. But I want to choose the lower precision type and prefer complex.

Reply:
Aha, thanks for taking a look. My mistake.